File: mobile/CloudProgram/cloudfunctions/get-work-items/WorkItem.ts
export class WorkItem {
  id: string = '';
  orgId: string = '';
  code: string = '';
  workItemType: string = '';
  originType: string = '';
  originId: string = '';
  originName: string = '';
  title: string = '';
  description: string = '';
  ownerId: string = '';
  ownerName: string = '';
  priority: string = 'medium';
  status: string = 'open';
  deadline: Date | null = null;
  slaDeadline: Date | null = null;
  escalationLevel: number = 0;
  completionCondition: string = '';
  sourceRuleId: string = '';
  sourceRuleName: string = '';
  version: number = 1;
  sourceType: string = 'manual';
  sourceId: string = '';
  correlationId: string = '';
  isDeleted: boolean = false;
  createdAt: Date | null = null;
  createdBy: string = '';
  updatedAt: Date | null = null;
  updatedBy: string = '';

  getClassName(): string {
    return 'WorkItem';
  }

  getFieldTypeMap(): Map<string, string> {
    const map = new Map<string, string>();
    map.set('id', 'String');
    map.set('orgId', 'String');
    map.set('code', 'String');
    map.set('workItemType', 'String');
    map.set('originType', 'String');
    map.set('originId', 'String');
    map.set('originName', 'String');
    map.set('title', 'String');
    map.set('description', 'String');
    map.set('ownerId', 'String');
    map.set('ownerName', 'String');
    map.set('priority', 'String');
    map.set('status', 'String');
    map.set('deadline', 'Date');
    map.set('slaDeadline', 'Date');
    map.set('escalationLevel', 'Integer');
    map.set('completionCondition', 'String');
    map.set('sourceRuleId', 'String');
    map.set('sourceRuleName', 'String');
    map.set('version', 'Integer');
    map.set('sourceType', 'String');
    map.set('sourceId', 'String');
    map.set('correlationId', 'String');
    map.set('isDeleted', 'Boolean');
    map.set('createdAt', 'Date');
    map.set('createdBy', 'String');
    map.set('updatedAt', 'Date');
    map.set('updatedBy', 'String');
    return map;
  }

  getPrimaryKeyList(): string[] {
    return ['id'];
  }

  getIndexList(): string[] {
    return ['id'];
  }

  getEncryptedFieldList(): string[] {
    return [];
  }

  static parseFrom(data: any): WorkItem {
    const obj = new WorkItem();
    if (data) {
      obj.id = data.id ?? '';
      obj.orgId = data.orgId ?? '';
      obj.code = data.code ?? '';
      obj.workItemType = data.workItemType ?? '';
      obj.originType = data.originType ?? '';
      obj.originId = data.originId ?? '';
      obj.originName = data.originName ?? '';
      obj.title = data.title ?? '';
      obj.description = data.description ?? '';
      obj.ownerId = data.ownerId ?? '';
      obj.ownerName = data.ownerName ?? '';
      obj.priority = data.priority ?? 'medium';
      obj.status = data.status ?? 'open';
      obj.deadline = data.deadline ? new Date(data.deadline) : null;
      obj.slaDeadline = data.slaDeadline ? new Date(data.slaDeadline) : null;
      obj.escalationLevel = data.escalationLevel ?? 0;
      obj.completionCondition = data.completionCondition ?? '';
      obj.sourceRuleId = data.sourceRuleId ?? '';
      obj.sourceRuleName = data.sourceRuleName ?? '';
      obj.version = data.version ?? 1;
      obj.sourceType = data.sourceType ?? 'manual';
      obj.sourceId = data.sourceId ?? '';
      obj.correlationId = data.correlationId ?? '';
      obj.isDeleted = data.isDeleted === true;
      obj.createdAt = data.createdAt ? new Date(data.createdAt) : null;
      obj.createdBy = data.createdBy ?? '';
      obj.updatedAt = data.updatedAt ? new Date(data.updatedAt) : null;
      obj.updatedBy = data.updatedBy ?? '';
    }
    return obj;
  }
}

File: mobile/Application/lib/models/work_item.dart
/// 统一工作项（P0-A）：审批/自动任务/项目任务/风险/数据治理统一抽象。
class WorkItem {
  const WorkItem({
    required this.id,
    required this.orgId,
    this.code = '',
    this.workItemType = '',
    this.originType = '',
    this.originId = '',
    this.originName = '',
    this.title = '',
    this.description = '',
    this.ownerId = '',
    this.ownerName = '',
    this.priority = 'medium',
    this.status = 'open',
    this.deadline,
    this.slaDeadline,
    this.escalationLevel = 0,
    this.completionCondition = '',
    this.sourceRuleId = '',
    this.sourceRuleName = '',
    required this.createdAt,
    this.updatedAt,
  });

  final String id;
  final String orgId;
  final String code;
  final String workItemType;
  final String originType;
  final String originId;
  final String originName;
  final String title;
  final String description;
  final String ownerId;
  final String ownerName;
  final String priority;
  final String status;
  final DateTime? deadline;
  final DateTime? slaDeadline;
  final int escalationLevel;
  final String completionCondition;
  final String sourceRuleId;
  final String sourceRuleName;
  final DateTime createdAt;
  final DateTime? updatedAt;

  bool get isOpen => status == 'open';
  bool get isDone => status == 'done';
  bool get isOverdue =>
      isOpen && slaDeadline != null && slaDeadline!.isBefore(DateTime.now());

  factory WorkItem.fromJson(Map<String, dynamic> json) => WorkItem(
        id: (json['id'] as String?) ?? '',
        orgId: (json['orgId'] as String?) ?? '',
        code: (json['code'] as String?) ?? '',
        workItemType: (json['workItemType'] as String?) ?? '',
        originType: (json['originType'] as String?) ?? '',
        originId: (json['originId'] as String?) ?? '',
        originName: (json['originName'] as String?) ?? '',
        title: (json['title'] as String?) ?? '',
        description: (json['description'] as String?) ?? '',
        ownerId: (json['ownerId'] as String?) ?? '',
        ownerName: (json['ownerName'] as String?) ?? '',
        priority: (json['priority'] as String?) ?? 'medium',
        status: (json['status'] as String?) ?? 'open',
        deadline: _toDate(json['deadline']),
        slaDeadline: _toDate(json['slaDeadline']),
        escalationLevel: (json['escalationLevel'] as num?)?.toInt() ?? 0,
        completionCondition: (json['completionCondition'] as String?) ?? '',
        sourceRuleId: (json['sourceRuleId'] as String?) ?? '',
        sourceRuleName: (json['sourceRuleName'] as String?) ?? '',
        createdAt: _toDate(json['createdAt']) ?? DateTime.now(),
        updatedAt: _toDate(json['updatedAt']),
      );
}

DateTime? _toDate(dynamic v) {
  if (v == null) return null;
  if (v is DateTime) return v;
  if (v is int) return DateTime.fromMillisecondsSinceEpoch(v);
  if (v is String) {
    final n = int.tryParse(v);
    if (n != null) return DateTime.fromMillisecondsSinceEpoch(n);
    return DateTime.tryParse(v);
  }
  return null;
}

const Map<String, String> kWorkItemTypeLabels = {
  'approval': '审批',
  'auto_task': '自动任务',
  'project_task': '项目任务',
  'risk': '风险整改',
  'data_quality': '数据治理',
  'compliance': '合规事项',
  'resolution': '决议执行',
};

String workItemTypeLabel(String type) => kWorkItemTypeLabels[type] ?? type;
